pkg/envoy/embedded_envoy.go

```diff
@@ -258,67 +263,65 @@ func startEmbeddedEnvoy(config embeddedEnvoyConfig) (*EmbeddedEnvoy, error) {
 }
 
 // newEnvoyLogPiper creates a writer that parses and logs log messages written by Envoy.
-func newEnvoyLogPiper() io.WriteCloser {
+func (o *onDemandXdsStarter) newEnvoyLogPiper() io.WriteCloser {
 	reader, writer := io.Pipe()
 	scanner := bufio.NewScanner(reader)
 	scanner.Buffer(nil, 1024*1024)
 	go func() {
-		scopedLog := log.WithFields(logrus.Fields{
-			logfields.LogSubsys: "unknown",
-			logfields.ThreadID:  "unknown",
-		})
-		level := "debug"
-
 		for scanner.Scan() {
 			line := scanner.Text()
-			var msg string
+
+			logThreadID := "unknown"
+			logLevel := "debug"
+			logSubsys := "unknown"
+			logMsg := ""
 
 			parts := strings.SplitN(line, "|", 4)
 			// Parse the line as a log message written by Envoy, assuming it
 			// uses the configured format: "%t|%l|%n|%v".
 			if len(parts) == 4 {
-				threadID := parts[0]
-				level = parts[1]
-				loggerName := parts[2]
+				logThreadID = parts[0]
+				logLevel = parts[1]
+				logSubsys = fmt.Sprintf("envoy-%s", parts[2])
 				// TODO: Parse msg to extract the source filename, line number, etc.
-				msg = fmt.Sprintf("[%s", parts[3])
-
-				scopedLog = log.WithFields(logrus.Fields{
-					logfields.LogSubsys: fmt.Sprintf("envoy-%s", loggerName),
-					logfields.ThreadID:  threadID,
-				})
+				logMsg = fmt.Sprintf("[%s", parts[3])
```

Comment:
is the prefix `[` intentional? :thinking:  (looks like this is there since the introduction of the embedded Envoy.

Reply:
probably we can just let it be right now :D